tag_release.py

```diff
@@ -0,0 +1,554 @@
+    # is it a branch?
+    branch_url = "https://api.github.com/repos/{repo}/branches/{branch}".format(
+        repo=repo_name,
+        branch=ref,
+    )
+    branch_resp = session.get(branch_url)
+    if branch_resp.ok:
+        branch = branch_resp.json()
+        commit = branch["commit"]["commit"]
+        return {
+            "ref": ref,
+            "ref_type": "branch",
+            "sha": branch["commit"]["sha"],
+            "message": commit["message"],
+            "author": commit["author"],
+            "committer": commit["committer"],
+        }
+
+    # is it a tag?
+    tag_url = "https://api.github.com/repos/{repo}/git/refs/tags/{tag}".format(
+        repo=repo_name,
+        tag=ref,
+    )
+    tag_resp = session.get(tag_url)
+    if tag_resp.ok:
+        tag = tag_resp.json()
+        # need to do a subsequent API call to get the tagged commit
+        commit_url = tag["object"]["url"]
+        commit_resp = session.get(commit_url)
+        if commit_resp.ok:
+            commit = commit_resp.json()
+            return {
+                "ref": ref,
+                "ref_type": "tag",
+                "sha": commit["sha"],
+                "message": commit["message"],
+                "author": commit["author"],
+                "committer": commit["committer"],
+            }
+
+    msg = "No commit for {ref} in {repo}".format(
+        ref=ref, repo=repo_name,
+    )
+    raise ValueError(msg)
+
+
+def get_latest_commit_for_parent_repo(
+        repo_name, parent_repo_name, parent_branch_name, session,
+    ):
+    if parent_repo_name == "edx/edx-platform":
+        requirements_file = "requirements/edx/github.txt"
+    else:
+        requirements_file = "requirements.txt"
+
+    req_file_url = "https://raw.githubusercontent.com/{parent_repo}/{branch}/{req_file}".format(
+        parent_repo=parent_repo_name,
+        branch=parent_branch_name,
+        req_file=requirements_file,
+    )
+    req_file_resp = session.get(req_file_url)
+    req_file_resp.raise_for_status()
+    req_file_lines = req_file_resp.text.splitlines()
+    # strip lines that are empty, or start with comments
+    relevant_lines = [line for line in req_file_lines
+                      if line and not line.isspace() and not line.startswith("#")]
+
+    # find the line that corresponds to this repo
+    repo_lines = [line for line in relevant_lines if repo_name in line]
+    if not repo_lines:
+        msg = "{repo_name} dependency not found in {parent} repo".format(
+            repo_name=repo_name, parent=parent_repo_name,
+        )
+        raise ValueError(msg)
+    if len(repo_lines) > 1:
+        msg = "multiple {repo_name} dependencies found in {parent} repo".format(
+            repo_name=repo_name, parent=parent_repo_name,
+        )
+        raise ValueError(msg)
+
+    dependency_line = repo_lines[0]
+    # parse out the branch/tag
+    match = REQUIREMENT_RE.search(dependency_line)
```

Comment:
Since you're matching with this anyway, it'd be good to parse the whole requirements.txt file into a list of match objects (or dicts derived from them), and then find the proper line with the structured data, instead of a simple substring match, which could find multiple lines where only one is correct.